docs/building-rpms-from-source.md

```diff
@@ -62,62 +61,76 @@ The prepare script `mkosi.prepare` then looks as follows:
 #!/bin/sh
 set -e
 
-if [ "$1" = "build" ]; then
-    DEPS="--buildrequires"
-else
-    DEPS="--requires"
+if [ "$1" = "final" ]; then
+    exit 0
 fi
 
 mkosi-chroot \
+    env --chdir=mkosi \
     rpmspec \
     --query \
-    "$DEPS" \
+    --buildrequires \
     --define "_topdir /var/tmp" \
-    --define "_sourcedir mkosi/rpm" \
-    mkosi/rpm/mkosi.spec |
-        grep -E -v mkosi |
-        xargs -d '\n' dnf install
-
-if [ "$1" = "build" ]; then
-    until mkosi-chroot \
-        env --chdir=mkosi \
-        rpmbuild \
-        -bd \
-        --build-in-place \
-        --define "_topdir /var/tmp" \
-        --define "_sourcedir rpm" \
-        --define "_build_name_fmt %%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm" \
-        rpm/mkosi.spec
-    do
-        EXIT_STATUS=$?
-        if [ $EXIT_STATUS -ne 11 ]; then
-            exit $EXIT_STATUS
-        fi
-
-        dnf builddep /var/tmp/SRPMS/mkosi-*.buildreqs.nosrc.rpm
-    done
-fi
+    --define "_sourcedir rpm" \
+    rpm/mkosi.spec | sort --unique >/tmp/buildrequires
+
+xargs --arg-file /tmp/buildrequires --delimiter '\n' mkosi-install
+
+until mkosi-chroot \
+    env --chdir=mkosi \
+    rpmbuild \
+    -bd \
+    --build-in-place \
+    --define "_topdir /var/tmp" \
+    --define "_sourcedir rpm" \
+    --define "_build_name_fmt %%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm" \
+    rpm/mkosi.spec
+do
+    EXIT_STATUS=$?
+    if [ $EXIT_STATUS -ne 11 ]; then
```

Comment:
I assume 11 is the return code for missing build requirements, a comment for that here in the docs (but not necessarily in the actual prepare script) would be nice.

Reply:
We already mention this below